octodns/provider/ns1.py

```diff
@@ -298,6 +309,34 @@ def __init__(self, id, api_key, retry_count=4, monitor_regions=None,
         self._client = Ns1Client(api_key, parallelism, retry_count,
                                  client_config)
 
+    # Allowed filter configurations:
+    # 1. Filter chain is the basic filter chain
+    # 2. In addition, it could have the 'geofence_country' filter and or
+    # 'geofence_regional' filter only at index 1 in the filter chain
+    # 3. If both country and regional filters are present, they are required
+    # to be as below:
+    #   - 'geofence_regional' at index 1
+    #   - 'geofence_country' at index 2
+    # Any other deviation is returned as an unsupported filter configuration
+    def _valid_filter_config(self, filter_cfg):
+        has_region = 'geofence_regional' in [f['filter'] for f in filter_cfg]
+        has_country = 'geofence_country' in [f['filter'] for f in filter_cfg]
+        return filter_cfg == self._get_updated_filter_chain(has_region,
+                                                            has_country)
+
+    def _get_updated_filter_chain(self, has_region, has_country):
+        filters = deepcopy(self._BASIC_DYNAMIC_FILTERS)
+        if has_region:
+            filters.insert(self._REGION_FILTER_INDEX, self._REGION_FILTER)
+        if has_country:
+            if has_region:
+                filters.insert(self._COUNTRY_FILTER_INDEX_WITH_REGION,
+                               self._COUNTRY_FILTER)
+            else:
+                filters.insert(self._COUNTRY_FILTER_INDEX,
+                               self._COUNTRY_FILTER)
```

Comment:
Might be worth exploring whether it'd be easier to follow if there were 4 lists: base, with_region, with_country, and with_country_and_region which would let you just compare agains the 4 and avoid the inserting and index stuff.

Reply:
> Might be worth exploring whether it'd be easier to follow if there were 4 lists: base, with_region, with_country, and with_country_and_region which would let you just compare agains the 4 and avoid the inserting and index stuff.

Fine by that, changed. Personally wasn't crazy about dealing with specific indices.